esphome/components/ld2450/ld2450.h

```diff
@@ -0,0 +1,281 @@
+  void set_throttle(uint16_t value) { this->throttle_ = value; };
+  void read_all_info();
+  void query_zone_info();
+  void restart_and_read_all_info();
+  void set_bluetooth(bool enable);
+  void set_multi_target(bool enable);
+  void set_baud_rate(const std::string &state);
+  void set_zone_type(const std::string &state);
+  void publish_zone_type();
+  void factory_reset();
+  uint16_t convert_seconds_to_ms(uint16_t value) { return value * 1000; };
+#ifdef USE_TEXT_SENSOR
+  void set_direction_text_sensor(int target, text_sensor::TextSensor *s);
+#endif
+#ifdef USE_NUMBER
+  void set_zone_coordinate(uint8_t zone);
+  void set_zone_x1_number(int zone, number::Number *n);
+  void set_zone_y1_number(int zone, number::Number *n);
+  void set_zone_x2_number(int zone, number::Number *n);
+  void set_zone_y2_number(int zone, number::Number *n);
+#endif
+#ifdef USE_SENSOR
+  void set_move_x_sensor(int target, sensor::Sensor *s);
+  void set_move_y_sensor(int target, sensor::Sensor *s);
+  void set_move_speed_sensor(int target, sensor::Sensor *s);
+  void set_move_angle_sensor(int target, sensor::Sensor *s);
+  void set_move_distance_sensor(int target, sensor::Sensor *s);
+  void set_move_resolution_sensor(int target, sensor::Sensor *s);
+  void set_zone_target_count_sensor(int zone, sensor::Sensor *s);
+  void set_zone_still_target_count_sensor(int zone, sensor::Sensor *s);
+  void set_zone_moving_target_count_sensor(int zone, sensor::Sensor *s);
+#endif
+  void reset_radar_zone();
+  void set_radar_zone(int32_t zone_type, int32_t zone1_x1, int32_t zone1_y1, int32_t zone1_x2, int32_t zone1_y2,
+                      int32_t zone2_x1, int32_t zone2_y1, int32_t zone2_x2, int32_t zone2_y2, int32_t zone3_x1,
+                      int32_t zone3_y1, int32_t zone3_x2, int32_t zone3_y2);
+
+ protected:
+  ESPPreferenceObject pref_;
+  int two_byte_to_int_(char firstbyte, char secondbyte) { return (int16_t) (secondbyte << 8) + firstbyte; }
+  void send_command_(uint8_t command_str, const uint8_t *command_value, int command_value_len);
+  void set_config_mode_(bool enable);
+  void handle_periodic_data_(uint8_t *buffer, int len);
+  bool handle_ack_data_(uint8_t *buffer, int len);
+  void process_zone_(uint8_t *buffer);
+  void readline_(int readch, uint8_t *buffer, int len);
+  void get_version_();
+  void get_mac_();
+  void query_zone_();
+  void restart_();
+  void send_set_zone_command_();
+  void convert_int_values_to_hex_(const int *values, uint8_t *bytes);
+  void save_to_flash_(float value);
+  float restore_from_flash_();
+  Target target_info_[MAX_TARGETS];
+  Zone zone_config_[MAX_ZONES];
+  int16_t decode_coordinate_(uint8_t low_byte, uint8_t high_byte) {
+    int16_t coordinate = (high_byte & 0x7F) << 8 | low_byte;
+    if ((high_byte & 0x80) == 0)
+      coordinate = -coordinate;
+    return coordinate;  // mm
+  }
+  int16_t decode_speed_(uint8_t low_byte, uint8_t high_byte) {
+    int16_t speed = (high_byte & 0x7F) << 8 | low_byte;
```

Comment:
Fixes target direction always being moving away. Thanks to @Adoney03 in the Apollo discord
```suggestion
    int16_t speed = (high_byte & 0x7F) << 8 | low_byte;
    if ((high_byte & 0x80) == 0)
       speed = -speed;
```